### PR TITLE
Match with filter in the list on update (#608)

### PR DESCRIPTION
**Pull request by sanbornsen**

### CODE REVIEW

File: src/app/work-item/work-item.service.ts [hunk 1]
```diff
@@ -198,8 +199,11 @@ export class WorkItemService {
     if (this.prevFilters.length) {
       for (let i = 0; i < this.prevFilters.length; i++) {
         // In case of assignee filter
-        if (this.prevFilters[i].id === 1 && this.prevFilters[i].active === true) {
-          if (workItem.relationalData.assignees.findIndex(item => item.id == this.prevFilters[i].value) === -1) {
+        if (this.prevFilters[i].id === 1
+            && this.prevFilters[i].active === true) {
+          if (typeof(workItem.relationships.assignees.data) === 'undefined' // If un-assigned
+              || workItem.relationships.assignees.data.findIndex(item => item.id == this.prevFilters[i].value) === -1 // If assignee is not current
```
Comment (debloper) [inline src/app/work-item/work-item.service.ts]: Try to evaluate the LHS before entering the conditional block, preferably just do the comparison within `( ... )`

I mean to say,

```
let assignees = workItem.relationships.assignees;

if (!assignees.data || assignees.data.findIndex(...) === -1) {
  ...
}
```

I think `!workItem.relationships.assignees.data` is vastly superior than `typeof(workItem.relationships.assignees.data) === 'undefined'` conditional and I also think the arrow function may not be apt/needed here, but I'm not 100% confident what are you trying to do with `item.id == this.prevFilters[i].value` comparison.

File: src/app/work-item/work-item.service.ts [hunk 2]
```diff
@@ -581,13 +586,25 @@ export class WorkItemService {
...
+            // Update work item attributes
+            this.workItems[updateIndex].attributes = updatedWorkItem.attributes;
+            this.workItems[updateIndex].relationships.assignees = updatedWorkItem.relationships.assignees;
+            this.workItems[updateIndex].relationships.creator = updatedWorkItem.relationships.creator;
+            this.workItems[updateIndex].relationships.baseType = updatedWorkItem.relationships.baseType;
+            // Resolve users for the updated item
+            this.resolveUsersForWorkItem(this.workItems[updateIndex]);
+          } else {
+            // Remove the item from the list
+            this.workItems.splice(updateIndex, 1);
```
Comment (debloper) [inline src/app/work-item/work-item.service.ts]: Do you rather mean to use `pop()`?

Comment (nimishamukherjee) [inline src/app/work-item/work-item.service.ts]: pop would remove the last element. We are using splice as the element could be anywhere on the list.

Comment (debloper) [inline src/app/work-item/work-item.service.ts]: Alright, makes sense then.

File: src/app/work-item/work-item.service.ts [hunk 3]
```diff
@@ -198,8 +199,11 @@ export class WorkItemService {
     if (this.prevFilters.length) {
       for (let i = 0; i < this.prevFilters.length; i++) {
         // In case of assignee filter
-        if (this.prevFilters[i].id === 1 && this.prevFilters[i].active === true) {
-          if (workItem.relationalData.assignees.findIndex(item => item.id == this.prevFilters[i].value) === -1) {
+        if (this.prevFilters[i].id === 1
+            && this.prevFilters[i].active === true) {
```
Comment (sanbornsen) [inline src/app/work-item/work-item.service.ts]: Did you mean !!this.prevFilters[i].id === true === !!this.prevFilters[i].active ?

**Comment by nimishamukherjee** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Pull request merged**